### PR TITLE
ci: add multi arch build and push

### PR DESCRIPTION
**Pull request by suhaspotluri**
Closes https://github.com/Bonial-International-GmbH/pod-image-swap-webhook/issues/107

### CODE REVIEW

**Review by pjungermann — APPROVED** [review]
LGTM. Thank you for your contribution!

I will leave the final review and merge to the team.

**Comment by pjungermann** [comment]
Closes issue #107

File: .github/workflows/publish-image.yml [hunk 1]
```diff
@@ -22,13 +22,20 @@ jobs:
...
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
 
-      - name: Login to ghcr.io
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-      - name: Push image
+      - name: Login to Docker Hub
```
Comment (martinohmann) [inline .github/workflows/publish-image.yml:31]:
```suggestion
      - name: Login to ghcr.io
```

File: .github/workflows/publish-image.yml [hunk 2]
```diff
@@ -22,13 +22,20 @@ jobs:
...
+        uses: docker/login-action@v3
+        with:
```
Comment (martinohmann) [inline .github/workflows/publish-image.yml:33]: This is missing `registry: ghcr.io`, we don't want to log in against Docker Hub 😉.

**Review by martinohmann — CHANGES_REQUESTED** [review]
Thanks for the PR @suhaspotluri!

I left some comments which need to be addressed before we can merge this.

**Pull request merged**